src/components/Question.jsx

```diff
@@ -93,6 +93,33 @@ export default class Question extends React.Component {
     this.props.onChange(this.props.index, newState);
   }
 
+  isDebugged(question) {
+    if (FormUtils.isRelevant(question) && !question[Constants.HAS_SUBQUESTION])
+      return false;
+    if (this.context.options.debugMode) return true;
+    return this.getSubQuestionsId(question);
+  }
+
+  getSubQuestionsId(question) {
+    const subQuestion = question[Constants.HAS_SUBQUESTION];
+    const startingQuestion = this.context.options.startingQuestionId;
+    let bool;
+
+    if (subQuestion && subQuestion.length > 0) {
+      for (let i = 0; i < subQuestion.length; i++) {
+        bool = JsonLdObjectUtils.checkId(subQuestion[i], startingQuestion);
+        if (bool) return bool;
+      }
+    }
+    if (!subQuestion) {
+      bool = JsonLdObjectUtils.checkId(question, startingQuestion);
+    }
+    if (question[Constants.IS_RELEVANT_IF]) {
```

Comment:
I thought that we want to separate traversal of question tree from checkId function ..

`getSubQuestions(Question question, boolean transitive, boolean reflexive)` -- this would just return array of all subquestions nothing more